### ISSUE TITLE
Log spamming.

### ISSUE THREAD

**Issue by stefan-jonasson** [balloob/pychromecast#60]
I get these in my logs every 5th second:

```
 ERROR (Thread-16) [pychromecast.socket_client] Failed to connect, retrying in 5s
Traceback (most recent call last):
  File "/home/stefan/.homeassistant/lib/pychromecast/socket_client.py", line 160, in initialize_connection
    self.socket.connect((self.host, 8009))
  File "/usr/lib/python3.4/ssl.py", line 846, in connect
    self._real_connect(addr, False)
  File "/usr/lib/python3.4/ssl.py", line 837, in _real_connect
    self.do_handshake()
  File "/usr/lib/python3.4/ssl.py", line 810, in do_handshake
    self._sslobj.do_handshake()
ConnectionResetError: [Errno 104] Connection reset by peer
```

I think it would be enough with an INFO level log instead?

**Comment by balloob** [comment]
Well no, it's an error that shouldn't happen. It's your chromecast hanging
up on us.

**Comment by stefan-jonasson** [comment]
I don't want to unplug my chromecast because it works. As I understand it I cannot disable the chromecast detection in the Home assistant discovery module so there is no way for me to stop this logging. Maybe we could add some try x times and then wait for some longer period and try again. 

Maybe just remove the traceback so it's not so verbose! 
If this is happening for me it's probably happening for other chromecasts as well.

**Comment by balloob** [comment]
I have no issues here. There is some max retries code available in pychromecast but we're not using it right now within Home Assistant.

As a quick solution for you: write a custom component that disables error logging for pychromecast

**Comment by balloob** [comment]
The traceback is helpful in debugging if any other error is happening. It is an error after all.

For HA, you could write a custom component to disable pychromecast error logging.

**Comment by ahocquet** [comment]
I had the same logs tonigh in my daemon script used to monitor chromecast. The problem is that after I got this "spam" logs, pychromecast is unable to do anything with my chromecast. I need to force quit my script and relaunch it.

Here's the logs : 

```
2015-12-14 03:29:58,579 - pychromecast.controllers - INFO - Receiver:Stopping current app 'E8C28D3C'
2015-12-14 04:37:01,222 - pychromecast.socket_client - ERROR - Error communicating with socket, resetting connection
2015-12-14 04:37:09,217 - pychromecast.socket_client - ERROR - Failed to connect, retrying in 5.000000s
Traceback (most recent call last):
  File "/usr/local/python3/lib/python3.4/site-packages/pychromecast/socket_client.py", line 213, in initialize_connection
    self.socket.connect((self.host, 8009))
  File "/usr/local/python3/lib/python3.4/ssl.py", line 841, in connect
    self._real_connect(addr, False)
  File "/usr/local/python3/lib/python3.4/ssl.py", line 832, in _real_connect
    self.do_handshake()
  File "/usr/local/python3/lib/python3.4/ssl.py", line 805, in do_handshake
    self._sslobj.do_handshake()
ConnectionResetError: [Errno 104] Connection reset by peer
2015-12-14 04:37:14,724 - pychromecast.controllers - INFO - Receiver:Updating status
2015-12-14 09:22:06,859 - pychromecast.socket_client - ERROR - Error reading from socket.
2015-12-14 09:22:06,860 - pychromecast.socket_client - ERROR - Error communicating with socket, resetting connection
```

**Comment by flyte** [comment]
This log spamming is affecting me too. All I did was switch my TV off (depowering the Chromecast) and I get this every five seconds:
ERROR:pychromecast.socket_client:Failed to connect, retrying in 5s
Traceback (most recent call last):
  File "/usr/local/lib/python3.4/site-packages/pychromecast/socket_client.py", line 160, in initialize_connection
    self.socket.connect((self.host, 8009))
  File "/usr/local/lib/python3.4/ssl.py", line 846, in connect
    self._real_connect(addr, False)
  File "/usr/local/lib/python3.4/ssl.py", line 833, in _real_connect
    socket.connect(self, addr)
OSError: [Errno 113] No route to host

Like @stefan-jonasson mentioned, it's not an exceptional circumstance when a Chromecast is not available on the network. In fact, the log message I'd expect to see is one message when the Chromecast goes offline, and another one when it's online again - the retries should be silent.

**Comment by Landrash** [comment]
Using my Chromecast in the same way as @flyte.
It's in general common practice to power the Chromecast from the TV/monitor or stereo system for the Chromecast audio.

**Comment by Jarod83** [comment]
I'm having the same problem after rebooting HA.
Eventually it works after a while.
Meanwhile HA frontend is not available.

**Comment by Esiravegna** [comment]
It is possible for the new version to actually cease to communicate with all chromecasts in the net? I've two of them, one offline, x.x.1.8, and another online , x.x.1.10. I see all of them as 'off' on HA, and I see a lot of these errors.
Thanks!

**Comment by Esiravegna** [comment]
I can confirm that replacing

```python
except socket.error:
```
by

```python
except (socket.error,OSError):
```
in [pychromecast](https://github.com/balloob/pychromecast/blob/master/pychromecast/socket_client.py#L225)
SEEMS to have solved the issue.

**Comment by rmkraus** [comment]
That is correct. This is fixed in the PR I have open.

On Tuesday, January 5, 2016, Esteban Siravegna <notifications@github.com>
wrote:

> I can confirm that replacing
>
> except socket.error:
>
> by
>
> except (socket.error,OSError):
>
> in pychromecast
> <https://github.com/balloob/pychromecast/blob/master/pychromecast/socket_client.py#L225>
> SEEMS to have solved the issue.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/balloob/pychromecast/issues/60#issuecomment-168996237>
> .
>


-- 
Thanks,

Ryan Kraus

**Comment by rmkraus** [comment]
All the issues that have been brought up here should now be fixed in the master branch.

**Comment by arsaboo** [comment]
@rmkraus Looks like the the [socket_client.py](ttps://github.com/balloob/pychromecast/blob/master/pychromecast/socket_client.py#L258) still has `except socket.error:` instead of `except (socket.error,OSError):`.

I ask this because I am inundated with `pychromecast.socket_client: Failed to connect, retrying in 5.000000s` in Home Assistant.

**Comment by rmkraus** [comment]
That error is purposefully thrown into the log as an Exception with traceback the first time it occurs while reconnecting. After the first failed attempt, the message will be thrown at the Debug level so as not to spam the log. It is an important message, so it is best not to silence it completely. The message will reappear in the log if it is raised the next time PyChromecast is reconnecting.

See: https://github.com/balloob/pychromecast/blob/bbcf72c3618a31a306ce85e4b7678b59277113b7/pychromecast/socket_client.py#L235

Since this was a year ago, I cannot 100% remember doing it. However, we did not choose to change the Exceptions that were trapped by the try-except clause. It ended up not being necessary. Instead, I implemented the behavior described above.
